connectors/soap/src/test/java/io/camunda/connectors/soap/xml/XmlToJsonMapperTest.java

```diff
@@ -0,0 +1,120 @@
+ */
+package io.camunda.connectors.soap.xml;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+public class XmlToJsonMapperTest {
+
+  @Test
+  void shouldParseSimpleXmlString() {
+    ToJsonMapper mapper = Mapper.toJson().build();
+    Map<String, Object> result = mapper.toJson("<foo>bar</foo>");
+    assertThat(result).contains(entry("foo", "bar"));
+  }
+
+  @Test
+  void shouldGenerateJsonWithoutNamespaces() throws JsonProcessingException {
+    ToJsonMapper mapper = Mapper.toJson().withPreserveNamespaces(false).build();
+    JsonNode json = mapper.toJson(parseXml("soap-message.xml"));
+    System.out.println(
+        new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(json));
+    assertTrue(json.has("Envelope"));
+    assertTrue(json.get("Envelope").has("Header"));
+    assertTrue(json.get("Envelope").has("Body"));
+  }
+
+  @Test
+  void shouldGenerateJsonWithNamespaces() throws JsonProcessingException {
+    ToJsonMapper mapper = Mapper.toJson().withPreserveNamespaces(true).build();
+    Document initialDoc = parseXml("soap-message.xml");
+    JsonNode json = mapper.toJson(initialDoc);
+    System.out.println(
+        new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(json));
+    assertTrue(json.has("soap:Envelope"));
+    assertTrue(json.get("soap:Envelope").has("soap:Header"));
+    assertTrue(json.get("soap:Envelope").has("soap:Body"));
+  }
+
+  @Test
+  void shouldCreateArrayForSameChildNodes() throws JsonProcessingException {
+    ToJsonMapper mapper = Mapper.toJson().build();
+    JsonNode json = mapper.toJson(parseXml("xml-list.xml"));
+    System.out.println(
+        new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(json));
+  }
+
+  private Document parseXml(String filename) {
+    DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+    documentBuilderFactory.setNamespaceAware(true);
+    try (InputStream in = getClass().getClassLoader().getResourceAsStream(filename)) {
+      return documentBuilderFactory.newDocumentBuilder().parse(in);
+    } catch (SAXException | IOException | ParserConfigurationException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @ParameterizedTest
```

Comment:
This is a test that generates output that I used in the readme.

The System.out.* lines can potentially be removed